pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -262,6 +264,8 @@ public void deleteSegmentFile() {
   private final boolean _isOffHeap;
   private final boolean _nullHandlingEnabled;
   private final SegmentCommitterFactory _segmentCommitterFactory;
+  // Indicating if the segment is waiting to be upload to segment stores.
+  private boolean _waitingForUploadToSegmentStore = false;
```

Comment:
When is this flag set to true ?

Reply:
It is set to be true in line 789 when the committer still needs to upload the segment to deep store.